snuba/subscriptions/scheduler.py

```diff
@@ -35,16 +40,16 @@ class TaskBuilder(ABC, Generic[TSubscription]):
 
     @abstractmethod
     def get_task(
-        self, subscription: Subscription, timestamp: int
-    ) -> Optional[ScheduledTask[TSubscription]]:
+        self, subscription_with_tick: SubscriptionWithTick, timestamp: int
+    ) -> Optional[ScheduledSubscriptionTask]:
```

Comment:
Unrelated. I wonder if we should move the subscription scheduling logic into arroyo.

Reply:
Curious why you say that. Do you have an actual use case in mind where scheduling would be useful outside Snuba?

Reply:
I think the idea of having a consumer that follows the watermark produced by another and then "does something" can be reused. Same for the synchronized consumer which could be used for the post processing as well.